### PR TITLE
Icchen/1397 export result

### PR DESCRIPTION
**Pull request by I-Chenn**
**Description**
This PR links to the sub-task of issue #1397: "Export results as txt files".
Two buttons are added to the "Fitting Result" panel and the "Full Log" panel respectively.

When there is no information on the info panel, the button doesn't show up:
image

When there's information on the panel and when the mouse cursor moves into the area, the button and a tooltip show up:
image

The button is able to export the content of the info panel:
image
and the default format of the output file name is like:
**ImageName-DateTime-2D_Fitting_Result**

Similar button can also be found on "Full Log" panel and the default format of the output file name is:
**ImageName-DateTime-2D_Fitting_Full_Log**

**Checklist**
- [X] changelog updated / ~no changelog update needed~
- [X] e2e test passing / ~~added corresponding fix~~
- [X] ~protobuf updated to the latest dev commit~ / no protobuf update needed

### CODE REVIEW

**Comment by I-Chenn** [comment]
@kswang1029 @YuHsuan-Hwang 
The minor configuration of the four commits after the pull request is due to the discussion with Yu-Hsuan, mostly related to the position of the export button.
I think the latest commits is OK for review now. Please take a look, thanks.

**Review by kswang1029 — CHANGES_REQUESTED** [review]
The tooltip may be simplified as "Export as txt". 

Screen Shot 2022-09-29 at 10 12 15

The +/- string is replaced by a unicode "±" in https://github.com/CARTAvis/carta-frontend/pull/1984 Would it still work when exporting as txt?

**Comment by I-Chenn** [comment]
> The +/- string is replaced by a unicode "±" in #1984 Would it still work when exporting as txt?

No, it also shows "+/-" string:
image

The export function simply export the exact same content on the output panel, which also shows "+/-" string:
image

It seems that the `AppStore.Instance.imageFittingStore.effectiveFrame.fittingLog` & `AppStore.Instance.imageFittingStore.effectiveFrame.fittingResult` is not influenced with the modification of #1984, I need to check for that.

Update:
@kswang1029, it seems that the `imageFittingStore` is still adopting "+/-" string for output, check for [this line](https://github.com/CARTAvis/carta-frontend/blob/dev/src/stores/ImageFittingStore.ts#L131) and [this line](https://github.com/CARTAvis/carta-frontend/blob/dev/src/stores/ImageFittingStore.ts#L134).
Should I correct it with the unicode you mentioned in this issue or should l leave it in #1984? (It seems that #1984 is not closed yet)

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-import {AnchorButton, Classes, Dialog, FormGroup, HTMLSelect, Icon, IDialogProps, Intent, NonIdealState, Position, Pre, Slider, Tab, Tabs, Text} from "@blueprintjs/core";
+import {AnchorButton, ButtonGroup, Classes, Dialog, FormGroup, HTMLSelect, Icon, IDialogProps, Intent, NonIdealState, Position, Pre, Slider, Tab, Tabs, Text} from "@blueprintjs/core";
 import {Tooltip2} from "@blueprintjs/popover2";
 import classNames from "classnames";
 import {AppStore, HelpType} from "stores";
 import {DraggableDialogComponent} from "components/Dialogs";
 import {SafeNumericInput} from "components/Shared";
 import {CustomIcon} from "icons/CustomIcons";
 import "./FittingDialogComponent.scss";
+import {exportTxtFile, getTimestamp} from "utilities";
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx:11]: Please move to the line before `import "./FittingDialogComponent.scss";`

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.scss [hunk 2]
```diff
@@ -48,6 +48,32 @@
         margin-bottom: 0;
     }
 
+    .output-result-button {
+        position: absolute;
+        bottom: 25px;
+        right: 25px;
+        margin: 0 0 0 0;
+        transition: opacity 400ms;
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/FittingDialog/FittingDialogComponent.scss]: Looks like the transition is not working.

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.scss [hunk 3]
```diff
@@ -48,6 +48,32 @@
...
+
+        .bp3-button {
+            transition: opacity 400ms;
+            user-select: none;
+        }
+    }
+
+    .output-log-button {
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/FittingDialog/FittingDialogComponent.scss]: Maybe we can combine `output-result-button` and `output-log-button` if their styles are the same.

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx [hunk 4]
```diff
@@ -33,6 +43,18 @@ export class FittingDialogComponent extends React.Component {
         return <SafeNumericInput value={isFinite(value) ? value : ""} placeholder={placeholder} onValueChange={onValueChange} buttonPosition="none" />;
     };
 
+    private exportResult = () => {
+        const content = AppStore.Instance.imageFittingStore.effectiveFrame?.fittingResult;
+        const fileName = `${AppStore.Instance.imageFittingStore.effectiveFrame?.filename}-${getTimestamp()}_2D-Fitting-Result`;
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx]: Shall we unify the suffix to either "_XX-XXX-XXX" or "-XX_XXX_XXX"?

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx [hunk 5]
```diff
@@ -60,12 +82,22 @@ export class FittingDialogComponent extends React.Component {
...
+                    </Tooltip2>
+                </ButtonGroup>
             </Pre>
         );
 
         const fullLogPanel = (
             <Pre className="fitting-result-pre">
                 <Text className="log-text">{fittingStore.effectiveFrame?.fittingLog ?? ""}</Text>
+                <ButtonGroup style={{opacity: this.isMouseEntered && fittingStore.effectiveFrame.fittingLog !== "" ? 1 : 0}}>
+                    <Tooltip2 content={"Exportput as txt"} position={Position.LEFT} className="output-log-button">
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx]: typo for `Export`

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
Comments for some minor issues:

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx (hunk 1 above)
Comment (I-Chenn) [reply src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx:11]: Done
image

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.scss (hunk 2 above)
Comment (I-Chenn) [reply src/components/Dialogs/FittingDialog/FittingDialogComponent.scss]: It's working now with the modification shown below:
image

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.scss (hunk 3 above)
Comment (I-Chenn) [reply src/components/Dialogs/FittingDialog/FittingDialogComponent.scss]: image
They're combined into ".output-button"

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx (hunk 4 above)
Comment (I-Chenn) [reply src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx]: It is now unified to "_XX-XXX-XXX"
image

File: src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx (hunk 5 above)
Comment (I-Chenn) [reply src/components/Dialogs/FittingDialog/FittingDialogComponent.tsx]: Sorry for the embarrassing error, it's now corrected:
image

**Review by YuHsuan-Hwang — APPROVED** [review]
New changes look good 👍

**Comment by YuHsuan-Hwang** [comment]
> The +/- string is replaced by a unicode "±" in #1984 Would it still work when exporting as txt?

@kswang1029 @I-Chenn #1984 also changed the log string in z profile fitting. Looks like the output txt file of z profile is working fine.

**Review by kswang1029 — APPROVED** [review]
this looks good. no regression is observed from e2e tests.

**Pull request merged**